File: 796-RotateString/796-RotateString.py
# Last updated: 5/3/2026, 12:58:15 PM
1class Solution:
2    def rotateString(self, s: str, goal: str) -> bool:
3        count = 0
4        while count < len(s):
5            rotated = s[len(s) - count:] + s[:len(s) - count]
6            if rotated == goal: return True
7            count += 1
8        return False

File: 796-RotateString/796-RotateString.cpp
// Last updated: 5/3/2026, 1:02:29 PM
1class Solution {
2public:
3    bool rotateString(string s, string goal) {
4        if (s.size() != goal.size()) return false;
5        string newString = s + s;
6        for (int i = 0; i < newString.size() - goal.size(); ++i) {
7            if (newString.substr(i, goal.size()) == goal) return true;
8        }
9        return false;
10    }
11};